python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -74,3 +83,118 @@ def get_node_info_by_output_name(self, output_name: str) -> Mapping[str, Any]:
     def get_node_info_by_asset_key(self, asset_key: AssetKey) -> Mapping[str, Any]:
         """Get a dbt node's dictionary representation in the manifest by its Dagster output name."""
         return self.node_info_by_asset_key[asset_key]
+
+
+@dataclass
+class DbtCliEventV2:
+    """Represents a dbt CLI event."""
+
+    event: Dict[str, Any]
+
+    @classmethod
+    def from_log(cls, log: str) -> "DbtCliEventV2":
+        """Parse an event according to https://docs.getdbt.com/reference/events-logging#structured-logging.
+
+        We assume that the log format is json.
+        """
+        event: Dict[str, Any] = json.loads(log)
+
+        return cls(event=event)
+
+    def __str__(self) -> str:
+        return self.event["info"]["msg"]
+
+
+@dataclass
+class DbtCliTask:
+    process: subprocess.Popen
+
+    def wait(self) -> Sequence[DbtCliEventV2]:
+        """Wait for the dbt CLI process to complete and return the events.
+
+        Returns:
+            Sequence[DbtCliEventV2]: A sequence of events from the dbt CLI process.
+        """
+        return list(self.stream())
+
+    def stream(self) -> Iterator[DbtCliEventV2]:
+        """Stream the events from the dbt CLI process.
+
+        Returns:
+            Iterator[DbtCliEventV2]: An iterator of events from the dbt CLI process.
+        """
+        for raw_line in self.process.stdout or []:
+            log: str = raw_line.decode().strip()
+            event = DbtCliEventV2.from_log(log=log)
+
+            yield event
+
+        # TODO: handle the return codes here!
+        # https://docs.getdbt.com/reference/exit-codes
+        return_code = self.process.wait()
+
+        return return_code
+
+
+class DbtClientV2(ConfigurableResource):
```

Comment:
Maybe just call this DbtClient, as the base DbtClient that DbtCliClient subclasses is not really doing anything (and can be renamed BaseDbtCliClient, _DbtClient, or just removed as its only child classes are DbtRpcClient and DbtCliClient, with the former being deprecated soon)

In the interest of minimal work, I think:

DbtClient -> _DbtClient
DbtClientV2 -> DbtClient

would be the easiest

Reply:
Ah and make sure to mark this as experimental